internal/pkg/log/logger.go

```diff
@@ -52,114 +58,94 @@ func (l *zapLogger) WithDuration(v time.Duration) Logger {
-	case "panic", "PANIC":
-		l.sugaredLogger.Desugar().Panic(message, fields...)
-	case "fatal", "FATAL":
-		l.sugaredLogger.Desugar().Fatal(message, fields...)
-	default:
-		l.sugaredLogger.Desugar().Info(message, fields...)
-	}
+func (l *zapLogger) Log(ctx context.Context, level, message string) {
+	l.logInLevel(level, l.message(ctx, message), l.fields(ctx)...)
 }
 
 func (l *zapLogger) Debugf(ctx context.Context, template string, args ...any) {
-	l.sugaredLogger.Desugar().Debug(
-		fmt.Sprintf(formatMessageUsingAttributes(template, ctxattr.Attributes(ctx)), args...),
-		l.prepareFields(ctx)...,
-	)
+	l.logger.Debug(l.template(ctx, template, args), l.fields(ctx)...)
 }
 
 func (l *zapLogger) Infof(ctx context.Context, template string, args ...any) {
-	l.sugaredLogger.Desugar().Info(
-		fmt.Sprintf(formatMessageUsingAttributes(template, ctxattr.Attributes(ctx)), args...),
-		l.prepareFields(ctx)...,
-	)
+	l.logger.Info(l.template(ctx, template, args), l.fields(ctx)...)
 }
 
 func (l *zapLogger) Warnf(ctx context.Context, template string, args ...any) {
-	l.sugaredLogger.Desugar().Warn(
-		fmt.Sprintf(formatMessageUsingAttributes(template, ctxattr.Attributes(ctx)), args...),
-		l.prepareFields(ctx)...,
-	)
+	l.logger.Warn(l.template(ctx, template, args), l.fields(ctx)...)
 }
 
 func (l *zapLogger) Errorf(ctx context.Context, template string, args ...any) {
-	l.sugaredLogger.Desugar().Error(
-		fmt.Sprintf(formatMessageUsingAttributes(template, ctxattr.Attributes(ctx)), args...),
-		l.prepareFields(ctx)...,
-	)
+	l.logger.Error(l.template(ctx, template, args), l.fields(ctx)...)
+}
+
+func (l *zapLogger) Logf(ctx context.Context, level, template string, args ...any) {
+	l.logInLevel(level, l.template(ctx, template, args), l.fields(ctx)...)
 }
 
 func (l *zapLogger) Sync() error {
-	return l.sugaredLogger.Sync()
+	return l.logger.Sync()
 }
 
 func (l *zapLogger) zapCore() zapcore.Core {
 	return l.core
 }
+
+func (l *zapLogger) logInLevel(level string, message string, fields ...zap.Field) {
+	switch level {
+	case "debug", "DEBUG":
+		l.logger.Debug(message, fields...)
+	case "info", "INFO":
+		l.logger.Info(message, fields...)
+	case "warn", "WARN":
+		l.logger.Warn(message, fields...)
+	case "error", "ERROR":
+		l.logger.Error(message, fields...)
+	case "dpanic", "DPANIC":
+		l.logger.DPanic(message, fields...)
+	case "panic", "PANIC":
+		l.logger.Panic(message, fields...)
+	case "fatal", "FATAL":
+		l.logger.Fatal(message, fields...)
+	default:
+		l.logger.Info(message, fields...)
+	}
+}
+
+func (l *zapLogger) message(ctx context.Context, message string) string {
+	var replacements []string
+	for _, kv := range ctxattr.Attributes(ctx).ToSlice() {
+		replacements = append(replacements, "<"+string(kv.Key)+">", kv.Value.Emit())
+	}
+	for _, kv := range l.attributes {
+		replacements = append(replacements, "<"+string(kv.Key)+">", kv.Value.Emit())
+	}
+	return strings.NewReplacer(replacements...).Replace(message)
```

Comment:
The both, `ctxattrs` and `logger.attributes` are used when formatting the message.